pydantic_core/core_schema.py

```diff
@@ -75,12 +98,34 @@ class ValidationInfo(Protocol):
     Argument passed to validation functions.
     """
 
-    data: Dict[str, Any]
-    """All of the fields and data being validated for this model."""
-    context: Dict[str, Any]
-    """Current validation context."""
-    config: CoreConfig | None
-    """The CoreConfig that applies to this validation."""
+    @property
```

Comment:
Having two types lets us minimize runtime logic since we can always return the last seen field. It's also a better user experience because users don't have to add an `assert info.field is not None` each time they want to use it.